subprojects/gradle/android/src/main/kotlin/com/avito/android/AndroidSdk.kt

```diff
@@ -56,18 +58,16 @@ class AndroidSdk(
             return ExistingDirectory.Impl(dir)
         }
 
-    private val apkSigner = ApkSigner.Impl(buildToolsPath, processRunner)
```

Comment:
The problem: build check plugin fails on a new project. 
It is caused by an implicit reading of `buildToolsVersion` system properly.
We could ask to add the system property. This is possible only in `settings.gradle` before plugins applying. But this is inconvenient and unexpected behavior. We don't need this version in build checks at all.

It seems that a better solution would be to get rid of system properties and environment variables, especially in common utilities.
They are hard for testing. Gradle test kit can override only [envs](https://docs.gradle.org/current/javadoc/org/gradle/testkit/runner/GradleRunner.html#withEnvironment-java.util.Map-), but not properties. In junit tests we have only bad options with dirty reflection or races in a real state. Wrapper for `System` class won't solve the problem with implicit dependency on properties.

Reply:
For unit tests, we can configure system properties in the test task.

Reply:
Technically, yes, we can. But it is a shared state for other tests. It's like a singleton on Gradle level.